packages/mds-geography/index.ts

```diff
@@ -1 +1,2 @@
 export { api } from './api'
+export { GeographyApiGetGeographyResponse, GeographyApiGetGeographiesResponse } from './types'
```

Comment:
I don't think this is actually the type you want. This is actually an _Express_ response object:
```ts
export type GeographyApiGetGeographyResponse = GeographyApiResponse<{
  data: { geographies: Geography[] | GeographySummary[] } | { geography: Geography | GeographySummary }
}>
```
I think the type you may want is just:
```ts
{
  data: { geographies: Geography[] | GeographySummary[] } | { geography: Geography | GeographySummary }
}
```
So perhaps, the change should be to declare as follows:
```ts
export type GeographyApiGetGeographyResponseBody = {
  data: { geographies: Geography[] | GeographySummary[] } | { geography: Geography | GeographySummary }
}

export type GeographyApiGetGeographyResponse = GeographyApiResponse<GeographyApiGetGeographyResponseBody>
```
With a similar change for _GeographyApiGetGeographiesResponse_ and then just export the body types:
```suggestion
export { GeographyApiGetGeographyResponseBody, GeographyApiGetGeographiesResponseBody } from './types'
```

Reply:
Ahh, yes that makes more sense. Good observation.

Funny thing, this was approached differently with Metrics response handling. With the addition of `ExtractApiResponseBody`:
```
/**
 * Extracts the body generic (B) for an APIResponse, in addition to possible error values.
 * Useful for frontend applications attempting to use the response payloads defined for Express.
 */
export declare type ExtractApiResponseBody<P> = P extends ApiResponse<infer T> ? T | ApiError : never;
```
We're extracting out the Body into a new type in the client instead:
```
type MetricsResponse = ExtractApiResponseBody<MetricsApiQueryResponse>
```

Either approach works. I'm not sure which one is the better though.

Reply:
They both work, but `ApiResponse` extends `express.Response` which means that at some level it's dependent on `Express` which feels like something that shouldn't be a concern outside of the backend. If there were something like `AxiosResponse<{Domain>` on the frontend that would feel equivalent to me. The `ApiResponse` type is going to have things like `res.locals`, etc.

Reply:
True true. But the client is pretty much always importing types now (`import type ...`) because of those unwanted and sometimes crashing dependencies, so it sort of gets around that.

Reply:
Except for the type contains `.locals` which you won’t have access to and the body type is _only_ available via inference. The specific body type feels better to me, but either approach is probably fine.